### PR TITLE
Update to TileDB 2.10.1

### PR DESCRIPTION
**Pull request by antalakas**

### CODE REVIEW

File: cpp/cmake/TileDB.cmake [hunk 1]
```diff
@@ -9,7 +9,18 @@ message(STATUS "start to set tiledb for version:${TILEDB_VERSION}")
 if(${TILEDB_VERSION} MATCHES "^([0-9]+)\\.([0-9]+)\\.([0-9]+)$")
 message(STATUS "start to set TILEDB_DOWNLOAD_URL and TILEDB_DOWNLOAD_SHA1")
 
-if(${TILEDB_VERSION} STREQUAL "2.9.4")
+if(${TILEDB_VERSION} STREQUAL "2.10.1")
```
Comment (Shelnutt2) [inline cpp/cmake/TileDB.cmake:12]: Why do we do matching on the version?

File: examples/TileDB.CSharp.Example/TileDB.CSharp.Example.csproj [hunk 2]
```diff
@@ -2,7 +2,7 @@
   <PropertyGroup>
     <OutputType>Exe</OutputType>
     <RootNamespace>TileDB.CSharp.Examples</RootNamespace>
-    <Version>5.0.3</Version>
+    <Version>5.0.4</Version>
```
Comment (Shelnutt2) [inline examples/TileDB.CSharp.Example/TileDB.CSharp.Example.csproj:5]: When we change TileDB minor version we must always change minor version of the API. This needs to be 5.1. This is what we do with TileDB-Cloud-REST, TileDB-Go and all other APIs.

File: sources/TileDB.CSharp/TileDB.CSharp.csproj [hunk 3]
```diff
@@ -2,7 +2,7 @@
 
   <PropertyGroup>
     <RootNamespace>TileDB.CSharp</RootNamespace>
-    <Version>5.0.3</Version>
+    <Version>5.0.4</Version>
```
Comment (Shelnutt2) [inline sources/TileDB.CSharp/TileDB.CSharp.csproj:5]: Bump minor version, should be 5.1.

File: sources/TileDB.CSharp/TileDB.CSharp.nuspec [hunk 4]
```diff
@@ -2,14 +2,14 @@
 <package >
   <metadata>
     <id>TileDB.CSharp</id>
-    <version>5.0.3</version>
+    <version>5.0.4</version>
```
Comment (Shelnutt2) [inline sources/TileDB.CSharp/TileDB.CSharp.nuspec:5]: Bump minor version, should be 5.1.

File: sources/TileDB.Interop/TileDB.Interop.csproj [hunk 5]
```diff
@@ -4,7 +4,7 @@
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
   </PropertyGroup>
   <PropertyGroup>
-    <Version>5.0.3</Version>
+    <Version>5.0.4</Version>
```
Comment (Shelnutt2) [inline sources/TileDB.Interop/TileDB.Interop.csproj:7]: Bump minor version, should be 5.1.

**Review by Shelnutt2 — CHANGES_REQUESTED** [review]

File: cpp/cmake/TileDB.cmake (hunk 1 above)
Comment (ihnorton) [reply cpp/cmake/TileDB.cmake:12]: This was carried over from the earlier API, there's definitions for the download targets for several versions below.

**Review by shaunrd0 — APPROVED** [review]

**Review by ihnorton — APPROVED** [review]

**Review by Shelnutt2 — APPROVED** [review]

**Pull request merged**